packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -241,7 +241,10 @@ export function createRootHooks<
     const queryClient = useQueryClient();
 
     const mutationKey = [path];
-    const defaultOpts = queryClient.getMutationDefaults(mutationKey);
+
+    const defaultOpts = queryClient.defaultMutationOptions(
+      queryClient.getMutationDefaults(mutationKey),
+    );
```

Comment:
cc @TkDodo 

this looks very odd to me, is this really how to fetch mutation options? it feels like we're fetching defaults twice

Reply:
queryClient.defaultMutationOptions can accept defaults as a parameter and it merge the parameters it accept with the global default giving precedence to the accepted parameters

Reply:
`queryClient.defaultMutationOptions` looks up the default options internally and merges with the passed options, so this shouldn't be necessary. What we do internally is this:

```
options: client.defaultMutationOptions(options),
```

so if you only want the default options, without passing anything, you'd need to do:

```
const defaultOpts = queryClient.defaultMutationOptions()
```

Reply:
> so if you only want the default options, without passing anything, you'd need to do:
> 
> ```
> const defaultOpts = queryClient.defaultMutationOptions()
> ```

i dont have much experience with react query, isn't this going to break users with defaults on the mutationKey?